orchagent/fdborch.cpp

```diff
@@ -36,6 +36,56 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
     Orch::addExecutor("FDB_NOTIFICATIONS", fdbNotifier);
 }
 
+/*
+ * sync up orchagent with libsai/ASIC for FDB entries.
+ *
+ * Currently NotificationProducer is used by syncd to inform FDB change,
+ * which means orchagent will miss the signal if it happens between orchagent shutdown and startup.
+ * Syncd doesn't know whether the signal has been lost or not.
+ * Also the source of notification event is from libsai/SDK.
+ *
+ * Syncd puts a copy of FDB into ASIC DB the moment it generates FDB notification.
+ * Here orchagent reads the data directly from ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY
+ */
+void FdbOrch::syncUpFdb()
+{
+    SWSS_LOG_ENTER();
+
+    string tableName = "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY";
+    DBConnector *db = new DBConnector(ASIC_DB, DBConnector::DEFAULT_UNIXSOCKET, 0);
+    swss::Table table(db, tableName);
+    vector<string> keys;
+
+    table.getKeys(keys);
+    for (const auto &key: keys)
+    {
+        sai_object_id_t bridge_port_id = SAI_NULL_OBJECT_ID;
+        sai_fdb_entry_type_t entryType = SAI_FDB_ENTRY_TYPE_STATIC;
```

Comment:
IMO, the default can by DYNAMIC and check for STATIC in _if_ statement down considering most of the FDBs learnt are dynamic